packages/grafana-data/src/dataframe/frameComparisons.ts

```diff
@@ -8,6 +8,9 @@ import { DataFrame } from '../types/dataFrame';
  * ```
  * areArraysEqual(a, b, framesHaveSameStructure);
  * ```
+ * NOTE: this does a shallow check on the FieldConfig properties, when using the query
+ * editor, this should be sufficient, however if applicaitons are mutating properties
+ * deep in the FieldConfig this will not recognize a change
```

Comment:
hmm, definite footgun. maybe the default should be deep diff and have a shallow flag if you know what you're doing?

Reply:
or i guess if you think shallow covers 80% of cases, then a deep flag instead?

Reply:
It should cover 100% of cases where a datasource or options are changed from the editor.  If we don't catch a change, the function should be updated so it does 😀